src/steps/10-FinancialDetails/CurrentlyHasFunding.spec.ts

```diff
@@ -28,4 +34,58 @@ describe("Testing CurrentlyHasFunding component", () => {
     });
   });
 
+  describe("Testing class functions", () => {
+    it("should retrieve user selection when getter currentData is called", async () => {
+      let selection = wrapper.vm.currentData;
+      expect(selection).toBe("");
+      await wrapper.setData({ "selectedHasFunding": "NO_FUNDING" });
+      await wrapper.vm.$nextTick();
+      selection = wrapper.vm.currentData;
+      expect(selection).toBe("NO_FUNDING");
+    });
+
+    it("should load data correctly on page load", async () => {
+      jest.spyOn(FinancialDetails, 'loadFundingRequirement').mockResolvedValue();
+      await wrapper.vm.loadOnEnter();
+
+      expect(FinancialDetails.loadFundingRequirement).toHaveBeenCalled();
+      expect(wrapper.vm.selectedHasFunding).toBe(FinancialDetails.hasFunding || "");
+    });
+
+    it("should save data correctly on page leave", async () => {
+      jest.spyOn(FinancialDetails, 'setHasFunding').mockResolvedValue();
+      jest.spyOn(FinancialDetails, 'saveFundingRequirement').mockResolvedValue();
+
+      wrapper.setData({ selectedHasFunding: "HAS_FUNDING" });
+
+      await wrapper.vm.saveOnLeave();
+
+      expect(FinancialDetails.setHasFunding).toHaveBeenCalledWith("HAS_FUNDING");
+      expect(FinancialDetails.saveFundingRequirement).toHaveBeenCalled();
+    });
+
+    it("should throw if data does not save on page leave", async () => {
+      jest.spyOn(wrapper.vm, 'hasChanged').mockReturnValue(true);
+      jest.spyOn(FinancialDetails, 'setHasFunding')
+        .mockImplementation(() => {throw new Error("Mock error")});
+      console.log = jest.fn();
+
+      await wrapper.vm.saveOnLeave();
+
+      expect(console.log).toHaveBeenCalledWith(new Error("Mock error"));
+    });
+
+    it("should navigate to RFD page if non-DITCO user", () => {
+      (AcquisitionPackage as any).isDitcoUser = jest.fn().mockReturnValue(false);
+
+      const pushMock = jest.fn();
+      wrapper.vm.$router = { push: pushMock } as any;
+
+      wrapper.vm.addNavigation();
+
+      expect(pushMock).toHaveBeenCalledWith({
+        name: routeNames.RFD,
+      });
```

Comment:
try not to use `any`
This code should work....
```
 it("should navigate to RFD page if non-DITCO user", () => {
      jest.spyOn(wrapper.vm, "isDitcoUser").mockReturnValue(false);

      const pushMock = jest.fn();
      wrapper.vm.$router = { push: pushMock as RawLocation } as VueRouter;
      wrapper.vm.addNavigation();

      expect(pushMock).toHaveBeenCalledWith({
        name: routeNames.RFD,
      });
    });
```

Reply:
Removed the `any` from the router but had to keep it on the `isDitcoUser` mock due to the way we export the function